### PR TITLE
Friendlier error messages when image path does not exist: Add warning message to imread()

### LINKED ISSUE #20167: Friendlier error messages when image path does not exist in cv2.seamlessClone()
##### System information (version)

- OpenCV => 4.2
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio Code 1.56.2

##### Detailed description
When running cv2.seamlessClone() the error is a bit misleading when the incorrect image path is supplied. It doesn't make it obvious that the problem is in the path

##### Steps to reproduce

```
import cv2
# Read images
DATA_PATH = "data_set/data"
source = cv2.imread(f"{DATA_PATH}/this image doesn't exist.png")
target = cv2.imread(f"{DATA_PATH}/10226_bg_27794.png")
swap = cv2.imread(f"{DATA_PATH}/10226_sw_27794_19200.png")
mask = cv2.imread(f"{DATA_PATH}/10226_mask_27794_19200.png")
# mask = mask/255
center = (112, 112)
output_normal = cv2.seamlessClone(swap, target, mask, center, cv2.NORMAL_CLONE)
output_mixed = cv2.seamlessClone(swap, target, mask, center, cv2.MIXED_CLONE)

# Output
---------------------------------------------------------------------------
error                                     Traceback (most recent call last)
<ipython-input-3-faf121550370> in <module>
     10 # mask = mask/255
     11 center = (112, 112)
---> 12 output_normal = cv2.seamlessClone(swap, target, mask, center, cv2.NORMAL_CLONE)
     13 output_mixed = cv2.seamlessClone(swap, target, mask, center, cv2.MIXED_CLONE)

error: OpenCV(4.5.1) C:\Users\appveyor\AppData\Local\Temp\1\pip-req-build-kh7iq4w7\opencv\modules\core\src\matrix.cpp:811: error: (-215:Assertion failed) 0 <= roi.x && 0 <= roi.width && roi.x + roi.width <= m.cols && 0 <= roi.y && 0 <= roi.height && roi.y + roi.height <= m.rows in function 'cv::Mat::Mat'
```

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues,
       forum.opencv.org, Stack Overflow, etc and have not found solution
   
 - [x] I updated to latest OpenCV version and the issue is still there
   
 - [x] There is reproducer code and related data files: videos, images, onnx, etc

### PR DESCRIPTION
**Pull request by NickJackolson**
resolves #20167  
Add a warning message using CV_LOG_WARNING() to imread().
Add a src check using CV_Assert() to seamlessClone().
These changes would help using imread() and seamlessClone(). Change in imread would especially help to new users when they can make mistakes with path.

Signed-off-by: nickjackolson <metedurlu@gmail.com>

<cut/>

System Info:
- Linux pop-os 5.13.0-7614-generic 14  x86_64 x86_64 x86_64 GNU/Linux
- cmake version 3.16.3
- g++ (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0

### Pull Request Readiness Checklist

- [ + ] I agree to contribute to the project under Apache 2 License.
- [ + ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ + ] The PR is proposed to proper branch
- [ + ] There is reference to original bug report and related work
- [ + ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ + ] The feature is well documented and sample code can be built with the project CMake

```
allow_multiple_commits=1
```

### CODE REVIEW

File: modules/imgcodecs/src/loadsave.cpp [hunk 1]
```diff
@@ -234,6 +234,7 @@ static ImageDecoder findDecoder( const String& filename ) {
     /// Open the file
     FILE* f= fopen( filename.c_str(), "rb" );
 
+    CV_LOG_IF_WARNING(NULL, !f, "image file could not be opened or path could not be found");
     /// in the event of a failure, return an empty image decoder
     if( !f )
```
Comment (alalek) [inline modules/imgcodecs/src/loadsave.cpp:239]: 1.

> `LOG_IF`

No need to use condition here, if two lines below we have the same condition.
So it makes sense to move message there without condition.

---

2.
Check [this line](https://github.com/opencv/opencv/blob/4.5.4/modules/imgcodecs/src/loadsave.cpp#L771) and dump "filename" value too.

---
3.

Original problem of mentioned issue is missing validation for input parameters in `seamlessClone()` implementation.
Check previous discussions #20617 and add `CV_Assert(!_src.empty());` to resolve the mentioned issue.

**Review by alalek — COMMENTED** [review]
Thank you for contribution!

File: modules/imgcodecs/src/loadsave.cpp (hunk 1 above)
Comment (NickJackolson) [reply modules/imgcodecs/src/loadsave.cpp]: First two suggestions, has been applied to first commit and third suggestion has been added as a separate commit.

**Comment by alalek** [comment]
Looks good!

This patch should go into [3.4 branch](https://github.com/opencv/opencv/wiki/Branches) first.
We will merge changes from 3.4 into 4.x regularly (weekly/bi-weekly).

Please:
- change "base" branch of this PR: 4.x => 3.4 (use "Edit" button near PR title)
- rebase your commits from 4.x onto 3.4 branch. For example:
    `git rebase -i --onto upstream/3.4 upstream/4.x`
    (check list of your commits, save and quit (Esc + "wq" + Enter)
    where `upstream` is configured by following [this GitHub guide](https://help.github.com/articles/configuring-a-remote-for-a-fork/) and fetched (`git fetch upstream`).
- push rebased commits into source branch of your fork (with `--force` option)

Note: no needs to re-open PR, apply changes "inplace".

**Comment by NickJackolson** [comment]
> Looks good!
> 
> This patch should go into [3.4 branch](https://github.com/opencv/opencv/wiki/Branches) first. We will merge changes from 3.4 into 4.x regularly (weekly/bi-weekly).
> 
> Please:
> 
> * change "base" branch of this PR: 4.x => 3.4 (use "Edit" button near PR title)
> * rebase your commits from 4.x onto 3.4 branch. For example:
>   `git rebase -i --onto upstream/3.4 upstream/4.x`
>   (check list of your commits, save and quit (Esc + "wq" + Enter)
>   where `upstream` is configured by following [this GitHub guide](https://help.github.com/articles/configuring-a-remote-for-a-fork/) and fetched (`git fetch upstream`).
> * push rebased commits into source branch of your fork (with `--force` option)
> 
> Note: no needs to re-open PR, apply changes "inplace".

Did as you instructed. `git log` now shows new commits coming after latest 3.4 commits. Waiting for checks(builds) to be completed.

**Review by alalek — APPROVED** [review]
Well done :+1:

**Pull request merged**